osu.Game/Modes/UI/HPDisplay.cs

```diff
@@ -0,0 +1,43 @@
+﻿using OpenTK;
+using OpenTK.Graphics;
+using osu.Framework.Configuration;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Graphics.Transformations;
+using System;
+
+namespace osu.Game.Modes.UI
+{
+    public class HPDisplay : Container
+    {
+        private Box background;
+        private Box fill;
+
+        public HPDisplay()
+        {
+            Children = new Drawable[]
+            {
+                background = new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = Color4.Gray,
+                },
+                fill = new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = Color4.White,
+                    Scale = new Vector2(0, 1),
+                }, 
+            };
+        }
+
+        public double Current;
+
+        public void Set(double value)

```

Comment:
I agree with @UselessToucan, this should just be a property and Set should cease to exist